libavfilter/vf_drawbox.c

```diff
@@ -140,11 +143,29 @@ static void draw_region(AVFrame *frame, DrawBoxContext *ctx, int left, int top,
     }
 }
 
+static enum AVFrameSideDataType box_source_string_parse(const char *box_source_string)
+{
+    av_assert0(box_source_string);
+    if (!strcmp(box_source_string, "side_data_detection_bbox")) {
+        return AV_FRAME_DATA_DETECTION_BBOXES;
+    } else {
+        return AVERROR(EINVAL);
+    }
+}
+
 static av_cold int init(AVFilterContext *ctx)
 {
     DrawBoxContext *s = ctx->priv;
     uint8_t rgba_color[4];
 
+    if (s->box_source_string) {
+        s->box_source = box_source_string_parse(s->box_source_string);
+        if ((int)s->box_source < 0) {
```

Comment:
did you verify if '(int)' is a must here?

Reply:
Yes, if no `(int)` here, the `if (s->box_source < 0)` will always be false even if s->box_source is AVERROR(EINVAL)